### PR TITLE
dovecot: Build a bottle for Linuxbrew: Depends on bzip2

### PR DESCRIPTION
**Pull request by maxim-belkin**

### CODE REVIEW

**Comment by maxim-belkin** [comment]
```
checking zlib.h usability... yes
checking zlib.h presence... yes
checking for zlib.h... yes
checking bzlib.h usability... no
checking bzlib.h presence... no
checking for bzlib.h... no
configure: error: Can't build with bzlib support: bzlib.h not found
```
need bzlib...

**Comment by maxim-belkin** [comment]
```
depends_on "bzip2" => :build unless OS.mac?
```

**Comment by sjackman** [comment]
```
checking bzlib.h usability... no
checking bzlib.h presence... no
checking for bzlib.h... no
configure: error: Can't build with bzlib support: bzlib.h not found
```
https://travis-ci.org/Linuxbrew/homebrew-core/builds/242773036#L1565
Depends on bzip2

**Comment by maxim-belkin** [comment]
?

**Comment by sjackman** [comment]
Sorry my page was stale. I hadn't seen your comment yet.

Looks like `bzip2` is also a run-time dependency:
```
==> brew linkage --test dovecot
System libraries:
  /lib/x86_64-linux-gnu/libbz2.so.1.0
  /lib/x86_64-linux-gnu/libc.so.6
  /lib/x86_64-linux-gnu/libcrypt.so.1
  /lib/x86_64-linux-gnu/libdl.so.2
  /usr/lib/x86_64-linux-gnu/libsqlite3.so.0
No broken dylib links
Unwanted system libraries:
  libbz2.so.1.0
  libsqlite3.so.0
==> FAILED
```
https://travis-ci.org/Linuxbrew/homebrew-core/builds/243027932#L2432
So…
```
depends_on "bzip2" unless OS.mac?
```

It appears to also depend on `sqlite`

File: Formula/dovecot.rb [hunk 1]
```diff
@@ -18,6 +18,8 @@ class Dovecot < Formula
 
   depends_on "openssl"
   depends_on "clucene" => :optional
+  depends_on "bzip2" unless OS.mac?
+  depends_on "sqlite"
```
Comment (sjackman) [inline Formula/dovecot.rb]: Should be…
```ruby
unless OS.mac?
  depends_on "bzip2"
  depends_on "sqlite"
end
```

Comment (sjackman) [inline Formula/dovecot.rb]: In `.travis.yml` and `circle.yml`, please remove `libsqlite0-dev` and `libsqlite3-dev`.

**Comment by maxim-belkin** [comment]
in a separate PR or this one?

**Comment by sjackman** [comment]
In this PR, please.

**Comment by maxim-belkin** [comment]
@sjackman do you suggest to remove libsqlite0 and libsqlite3 permanently or omit this commit when I do the merge? (Feel free to merge this PR yourself, if you like)

**Comment by sjackman** [comment]
Yes, remove the `-dev` packages permanently, `libsqlite0-dev` and `libsqlite3-dev`.

File: .travis.yml [hunk 2]
```diff
@@ -45,6 +45,8 @@ before_install:
       mysql-common
       unixodbc-dev
       zlib1g-dev
+      libsqlite0-dev
+      libsqlite3-dev
```
Comment (sjackman) [inline .travis.yml]: Please insert these lines in alphabetical order.

File: circle.yml [hunk 3]
```diff
@@ -61,6 +61,8 @@ dependencies:
         mysql-common
         unixodbc-dev
         zlib1g-dev
+        libsqlite0-dev
+        libsqlite3-dev
```
Comment (sjackman) [inline circle.yml]: Please insert these lines in alphabetical order.

File: .travis.yml (hunk 2 above)
Comment (maxim-belkin) [inline .travis.yml]: ok

**Comment by sjackman** [comment]
This TravisCI error was transient.
```
Error: Failed to download resource "readline"
```
https://travis-ci.org/Linuxbrew/homebrew-core/builds/244942873#L969

@maxim-belkin Would you like to rebase this PR onto `master`?

**Comment by maxim-belkin** [comment]
rebased and force-pushed per request in https://github.com/Linuxbrew/homebrew-core/pull/2930#issuecomment-314823216

**Comment by sjackman** [comment]
Looks good to me. Thanks, Maxim.

**Comment by sjackman** [comment]
Merged!

**Pull request closed without merging**